res/xml/dvorak.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
+<keyboard>
+	<row>
+		<key key0="backspace" key1="esc" key2="`" key3="~" key4="delete" width="1.5" />
+		<key key0="shift" width="1.5" />
+		<key key0="p" key1="accent_ring" />
+		<key key0="y" key1="accent_grave" key3="accent_aigu" />
+		<key key0="f" key2="-" key3="_" />
+		<key key0="g" key2="=" key3="+" />
+		<key key0="c" key1="accent_trema" key2="accent_circonflexe" key3="{" key4="}" />
+		<key key0="r" key3="[" key4="]" />
+		<key key0="l" key2="\\" key3="|" />
+	</row>
+	<row>
+		<key key0="a" key1="tab" key2="1" key3="!" />
+		<key key0="o" key2="2" key3="\@" />
+		<key key0="e" key2="3" key3="\#" />
+		<key key0="u" key2="4" key3="$" />
+		<key key0="i" key2="5" key3="%" />
+		<key key0="d" key2="6" key3="^" />
+		<key key0="h" key2="7" key3="&amp;" />
+		<key key0="t" key2="8" key3="*" />
+		<key key0="n" key2="9" key3="(" key4=")" />
+		<key key0="s" key2="0" />
+	</row>
+	<row>
+		<key key0="q" shift="0.5" />
+		<key key0="j" />
+		<key key0="k" />
+		<key key0="x" key1="accent_cedille" key2="." key3="&lt;" />
+		<key key0="b" key2="," key3="&gt;" />
+		<key key0="m" key2="/" key3="\?" />
+		<key key0="w" key1="accent_tilde" key2=";" key3=":" />
```

Comment:
Shouldn't `;` and `:` be on the `q` ?
Similarly:
- `_` and `-` should be on `z` instead of `f`
- `|` and `\\` on `s` instead of `l`
- `+` and `=` on `l`

Reply:
If you want them to be closest to their normal-keyboard equivalents, then yes, but with the far corners being harder to manipulate, I'm still unsure about the symbols I left on L, since I keep ending up with a pipe instead of an L. I think this might be to do with the screen having a high DPI though. Even with the lowest precision, it's all too easy to make mistakes.

If we wanted closest-equivalents, but without using the edge keys too much, it might work.
Putting `,`/`<` on P and `.`/`>` on Y would better mirror how the normal Dvorak keyboard looks.
There's not much room to put `'`/`"` nearby, but those could always be placed on J instead, with Q having `;`/`:`.
I'm not overly fond of using S, since it's on the very edge already. However, with Z free and a half-space from the edge, it's a good candidate for `_`/`-`.
With the backspace moved over to the right, L won't be on the edge, so `+`/`=` might be okay there?
`|` and `\` are a little trickier. I feel like there'd be room somewhere along the bottom row, though.

Reply:
Avoiding the edge keys in some cases is a good idea. It's OK to have some keys far from where they are on a full-size keyboard but I wonder if future Dvorak users will complain.

It's worth investigating the backspace on the right but if that doesn't work for you, let's keep it on the left.

Reply:
`,`/`<` and `.`/`>` look like important landmark of the layout so they might currently be a little too far down. If something have to be moved to the third row, I'd say `+`/`=` is the less frequent and less important.
`_`/`-` should be close to its real position because it's frequent and needs to be typed fast in the middle of a word.

Reply:
Sounds like a good plan. I'll make those changes tomorrow and give it a test.